prow/plugins/jira/jira.go

```diff
@@ -17,26 +17,45 @@ limitations under the License.
 package jira
 
 import (
-	"errors"
+	"context"
+	"encoding/json"
 	"fmt"
 	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 	"sync"
 
 	"github.com/andygrunwald/go-jira"
+	githubql "github.com/shurcooL/githubv4"
 	"github.com/sirupsen/logrus"
+	"github.com/trivago/tgo/tcontainer"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/github"
 	jiraclient "k8s.io/test-infra/prow/jira"
+	"k8s.io/test-infra/prow/labels"
 	"k8s.io/test-infra/prow/pluginhelp"
 	"k8s.io/test-infra/prow/plugins"
 )
 
 const (
-	PluginName = "jira"
+	PluginName            = "jira"
+	bugLink               = `[Jira Issue %s](%s/browse/%s)`
+	criticalSeverity      = "Critical"
+	importantSeverity     = "Important"
+	moderateSeverity      = "Moderate"
+	lowSeverity           = "Low"
+	informationalSeverity = "Informational"
+)
+
+var (
+	titleMatch           = regexp.MustCompile(`(?i)OCPBUGS-([0-9]+):`)
```

Comment:
This does not look like something that should live in the `kubernetes/test-infra` repository. This plugin should either be 

- configurable enough that anyone could use it to implement lifecycle management for their own setup
- a custom plugin deployed elsewhere, code not living in this repo

Reply:
Thanks for the feedback. A new repo under the `openshift-eng` organization is being created to house an external `jira-lifecycle` prow plugin. I will close this PR and create a new one that just contains some required improvements to the jira client instead.